### PR TITLE
Feature/gnps metadata table

### PR DESCRIPTION
**Pull request by axelwalter**
## Description

Some refactoring on GNPS/IonIdentityMolecularNetworking. The generation of the feature quantification table is not in IIMN anymore, but has a separate class GNPSQuantificationFile. Also a small class was added for the generation of a meta value
table GNPSMetaValueFile.

## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes. (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out 
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. If you click in the column that lists the failed tests you will get detailed error messages.
- Or click on the action: e.g., for clang-format linting

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

**Comment by axelwalter** [comment]
I adopted the name from the GNPSMGFile, but totally agree that store would be a much better name. Will change it for all of them.

**Review by enetz — APPROVED** [review]

File: src/openms/include/OpenMS/FORMAT/GNPSMetaValueFile.h [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+#include <OpenMS/DATASTRUCTURES/StringListUtils.h>
+
+#pragma once
+
+namespace OpenMS
+{
+  class OPENMS_DLLAPI GNPSMetaValueFile
+  {
+    public:
+      /// Generate a meta value table (tsv file) for GNPS FBMN with information on the input mzML files.
```
Comment (enetz) [inline src/openms/include/OpenMS/FORMAT/GNPSMetaValueFile.h:43]: can you add doxygen documentation to these new classes and add some more detail? e.g. is this an OpenMS internal file format, or is it used in external tools? For keeping compatibility in the future, it should be explained how and where the format is defined.

**Comment by timosachsenberg** [comment]
this one shows a lot of changes... do you need to merge develop into this branch?

File: src/openms/include/OpenMS/FORMAT/GNPSMetaValueFile.h (hunk 1 above)
Comment (axelwalter) [reply src/openms/include/OpenMS/FORMAT/GNPSMetaValueFile.h:43]: good idea, added the docs in the latest commits

**Comment by axelwalter** [comment]
because of the merge issues this PR got replaced by #6152

**Pull request closed without merging**